### PR TITLE
SSCS-10277-Adding HMC delete endpoint

### PR DESCRIPTION
**Pull request by ThomasKKC**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10277

### Change description ###
Implement call to HMC Delete Endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by wmarshid — APPROVED** [review]
I think this looks okay but would like someone more familiar with the pact tests to review.

File: src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/HearingDeleteConsumerTest.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    }
+
+    @Pact(consumer = CONSUMER_NAME)
+    public RequestResponsePact badRequestErrorFromDeleteHearing(PactDslWithProvider builder) {
+        return builder.given(CONSUMER_NAME
+                                 + " throws bad request error while trying to delete hearing")
+                .uponReceiving("Request to DELETE hearing for bad hearing request")
+                    .path(PATH_HEARING).method(HttpMethod.DELETE.toString())
+            .query(ID + "=" + VALID_CASE_ID)
+            .body(toJsonString(generateInvalidHearingRequest())).headers(headers)
```
Comment (dattipoeHMCTS) [inline src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/HearingDeleteConsumerTest.java:64]: Won't an invalid delete request be one that's missing the cancellation reason?

File: src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/HearingDeleteConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    @PactTestFor(pactMethod = "deleteHearingRequestForValidRequest")
+    public void shouldSuccessfullyDeleteHearingRequest() {
+        HearingResponse hearingResponse = hmcHearingApi.deleteHearingRequest(
+            IDAM_OAUTH2_TOKEN,
+            SERVICE_AUTHORIZATION_TOKEN,
+            VALID_CASE_ID,
+            generateHearingDeleteRequest()
+        );
+
+        assertTrue(hearingResponse.getHearingRequestId() > ZERO_LENGTH);
```
Comment (dattipoeHMCTS) [inline src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/HearingDeleteConsumerTest.java:135]: Can't you use not null and more concrete assertions?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java [hunk 3]
```diff
@@ -25,4 +27,13 @@ HearingResponse createHearingRequest(
         @RequestBody @Valid HearingRequestPayload hearingPayload
     );
 
+    @SuppressWarnings("PMD.UseObjectForClearerAPI")
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:30]: It probably is worth creating a holder object for the request.

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

**Review by wmarshid — APPROVED** [review]

**Pull request merged**